apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -110,7 +110,10 @@ public boolean canRunDataLayoutStrategyGeneration(TableMetadata tableMetadata) {
    */
   public boolean canRunDataCompaction(TableMetadata tableMetadata) {
     GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response) && isPrimaryTable(response);
+    return response != null
+        && checkCreationTimeEligibility(response)
+        && isPrimaryTable(response)
+        && (response.getTimePartitioning() != null || response.getClustering() != null);
```

Comment:
can response.getTimePArtitioning or getClustering be empty brackets?

Reply:
we only need to check partitioning, it should be either non-empty object or null, afaiu

Reply:
in that case should we add a check for non-empty object?